src/test/java/run/halo/app/controller/AdminControllerTest.java

```diff
@@ -0,0 +1,203 @@
+package run.halo.app.controller;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.nio.charset.StandardCharsets;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import run.halo.app.controller.admin.api.AdminController;
+import run.halo.app.core.ControllerExceptionHandler;
+import run.halo.app.model.dto.EnvironmentDTO;
+import run.halo.app.model.dto.LoginPreCheckDTO;
+import run.halo.app.model.entity.User;
+import run.halo.app.model.params.LoginParam;
+import run.halo.app.model.params.ResetPasswordParam;
+import run.halo.app.model.params.ResetPasswordSendCodeParam;
+import run.halo.app.security.token.AuthToken;
+import run.halo.app.service.AdminService;
+import run.halo.app.utils.JsonUtils;
+
+/**
+ * Admin controller test.
+ *
+ * @author guqing
+ * @date 2022-02-11
+ */
+@ExtendWith(MockitoExtension.class)
```

Comment:
If we just test controller only, please refer to https://docs.spring.io/spring-boot/docs/2.5.9/reference/html/features.html#features.testing.spring-boot-applications.spring-mvc-tests